test/react-native/features/steps/react-native-steps.rb

```diff
@@ -1,68 +1,99 @@
-When("I run {string}") do |event_type|
+def wait_for_true
+  # max_attempts = 300
+  max_attempts = 100
+  attempts = 0
+  assertion_passed = false
+  until (attempts >= max_attempts) || assertion_passed
+    attempts += 1
+    assertion_passed = yield
+    sleep 0.1
+  end
+  # raise 'Assertion not passed in 30s' unless assertion_passed
+end
+
+When('I run {string}') do |event_type|
   steps %Q{
     Given the element "scenario_name" is present within 60 seconds
     When I clear and send the keys "#{event_type}" to the element "scenario_name"
+    And I click the element "start_bugsnag"
     And I click the element "run_scenario"
   }
 end
 
-When("I run {string} and relaunch the app") do |event_type|
+When('I run {string} and relaunch the crashed app') do |event_type|
   steps %Q{
     When I run "#{event_type}"
     And I clear any error dialogue
-    And I relaunch the app
+    And I relaunch the app after a crash
   }
 end
 
-When("I relaunch the app") do
-  # This step should only be used when the app has crashed, but the notifier needs a little
-  # time to write the crash report before being forced to reopen.  From trials, 2s was not enough.
-  sleep(5)
+Then('the app is not running') do
+  wait_for_true do
```

Comment:
I think this could be:

```ruby
Maze::Wait.until do
  # stuff ...
end
```

then you don't need `wait_for_true`

Reply:
Good thinking, I'll factor that in when I take the next look at it.